### ISSUE TITLE
Sphinx 7.2.x - added `--import-mode=importlib` in pytest which makes it impossible to discover tests

### ISSUE THREAD

**Issue by befeleme** [sphinx-doc/sphinx#11740]
### Describe the bug

In the downstream RPM build for Fedora Linux, when I build Sphinx 7.2.6, I get errors on tests collection (see traceback below).
The culprit is in the pytest configuration added which uses `--import-mode=importlib`, added in https://github.com/sphinx-doc/sphinx/pull/11577.
According to the pytest's documentation, one drawback of this mode is ["that test modules are non-importable by each other"](https://docs.pytest.org/en/stable/explanation/pythonpath.html) which is the case in Sphinx.

The project's test suite here works because tests are invoked via `python -m pytest` which adds cwd to `sys.path`. For the downstream packaging we want to make sure the just-built-and-installed library is tested rather than the one in the current directory.

I don't know the motivation behind adding this config, but would you be wiling to remove it? I can send a patch.

```
+ /usr/bin/pytest -k 'not linkcheck and not test_latex_images and not test_build_latex_doc'
============================= test session starts ==============================
platform linux -- Python 3.12.0, pytest-7.4.2, pluggy-1.3.0
libraries: Sphinx-7.2.6, docutils-0.20.1
base tmp_path: /tmp/pytest-of-mockbuild/pytest-0
rootdir: /builddir/build/BUILD/sphinx-7.2.6
configfile: pyproject.toml
testpaths: tests
collected 1954 items / 3 errors / 8 deselected / 1946 selected

==================================== ERRORS ====================================
________________ ERROR collecting tests/test_build_linkcheck.py ________________
ImportError while importing test module '/builddir/build/BUILD/sphinx-7.2.6/tests/test_build_linkcheck.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/test_build_linkcheck.py:31: in <module>
    from .utils import CERT_FILE, http_server, https_server
E   ModuleNotFoundError: No module named 'tests.utils'; 'tests' is not a package
________________ ERROR collecting tests/test_ext_intersphinx.py ________________
ImportError while importing test module '/builddir/build/BUILD/sphinx-7.2.6/tests/test_ext_intersphinx.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/test_ext_intersphinx.py:22: in <module>
    from .test_util_inventory import inventory_v2, inventory_v2_not_having_version
E   ModuleNotFoundError: No module named 'tests.test_util_inventory'; 'tests' is not a package
____________ ERROR collecting tests/test_ext_napoleon_docstring.py _____________
ImportError while importing test module '/builddir/build/BUILD/sphinx-7.2.6/tests/test_ext_napoleon_docstring.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/test_ext_napoleon_docstring.py:21: in <module>
    from .ext_napoleon_pep526_data_google import PEP526GoogleClass
E   ModuleNotFoundError: No module named 'tests.ext_napoleon_pep526_data_google'; 'tests' is not a package
=========================== short test summary info ============================
ERROR tests/test_build_linkcheck.py
ERROR tests/test_ext_intersphinx.py
ERROR tests/test_ext_napoleon_docstring.py
!!!!!!!!!!!!!!!!!!! Interrupted: 3 errors during collection !!!!!!!!!!!!!!!!!!!!
======================= 8 deselected, 3 errors in 2.65s ========================
```

### How to Reproduce

In an active virtual environment with Python 3.11, code from this repository:

```bash
$ git checkout v7.2.6
$ pip install .[test]
$ pytest

### Environment Information

```text
Platform:              linux; (Linux-6.5.5-200.fc38.x86_64-x86_64-with-glibc2.37)
Python version:        3.11.6 (main, Oct  3 2023, 00:00:00) [GCC 13.2.1 20230728 (Red Hat 13.2.1-1)])
Python implementation: CPython
Sphinx version:        7.2.6
Docutils version:      0.19
Jinja2 version:        3.0.3
Pygments version:      2.14.0
```

**Comment by picnixz** [comment]
I think we added this to prevent any possible pollution of the system modules by accident. However, it clearly shows that there is an issue. 

So, either we patch the imports so that they are correct (because it appears that pytest assumes that thoses util modules are actually tests) or we revert the commit. 

@AA-Turner Can you tell us the rationale behind the change?

**Comment by AA-Turner** [comment]
Unintentional. Happy to revert.

A

**Comment by AA-Turner** [comment]
Resolved in 134703192d411bc08b17a5d06f1292c8c92a4ead
